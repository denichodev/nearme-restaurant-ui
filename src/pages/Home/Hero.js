import React from 'react';
import { shape, bool, arrayOf, object } from 'prop-types';
import { Link } from 'react-router-dom';

import { Row, Col } from '../../components/Layout';
import Rating from '../../components/Rating';
import s from './styles.css';

class Hero extends React.Component {
  renderTopFavorite = () => {
    const { loading, data } = this.props.favorite;

    if (loading) {
      return (
        <Col xs={12} sm={12} md={5}>
          <div className={s.top} />
        </Col>
      );
    }

    if (data.length) {
      const restaurant = data[0];

      return (
        <Col xs={12} sm={12} md={5}>
          <div className={s.top}>
            <img src={restaurant.cover} alt={restaurant.name} />
            <Link to={`/r/${restaurant.slug}`}>
              <div className={s.wrapper}>
                <Rating big point={restaurant.rating} />
                <div className={s.cuisine}>{restaurant.cuisine}</div>
                <div className={s.name}>{restaurant.name}</div>
                <div className={s.address}>{restaurant.location}</div>
              </div>
            </Link>
          </div>
        </Col>
      );
    }

    return null;
  };

  renderTopNine = () => {
    const { loading, data } = this.props.favorite;

    if (loading) {
      const loadingChildren = new Array(9);

      return loadingChildren.map(x => (
        <Col key={x} xs={6} sm={6} md={4}>
          <div className={s.top9} />
        </Col>
      ));
    }

    if (data.length) {
      return data.slice(1, 10).map((restaurant, index) => {
        const colSize = [2, 3, 8].includes(index) ? 12 : 6;

        return (
          <Col xs={colSize} sm={colSize} md={4} key={index}>
            <div className={s.topNine}>
              <img src={restaurant.cover} alt={restaurant.name} />
              <Link to={`/r/${restaurant.slug}`}>
                <div className={s.wrapper}>
                  <Rating point={restaurant.rating} />
                  <div className={s.cuisine}>{restaurant.cuisine}</div>
                  <div className={s.name}>{restaurant.name}</div>
                  <div className={s.address}>{restaurant.location}</div>
                </div>
              </Link>
            </div>
          </Col>
        );
      });
    }

    return null;
  };

  render() {
    return (
      <Row>
        <Col xs={12} sm={12}>
          <h1 className={s.popularHeading}>Popular</h1>
        </Col>
        {this.renderTopFavorite()}
        <Col xs={12} sm={12} md={7}>
          <Row>{this.renderTopNine()}</Row>
        </Col>
      </Row>
    );
  }
}

Hero.propTypes = {
  favorite: shape({
    loading: bool.isRequired,
    data: arrayOf(object).isRequired,
  }),
};

export default Hero;
