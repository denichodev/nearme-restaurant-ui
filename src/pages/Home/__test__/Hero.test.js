import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Hero from '../Hero';

describe('<Hero />', () => {
  it('renders correctly', () => {
    const props = {
      favorite: {
        loading: false,
        data: [
          {
            additionalInfo: [],
            address:
              'Gandaria City, Lantai Upper Ground, Jl. Sultan Iskandar Muda, Gandaria, Jakarta',
            cover: 'http://localhost:8080/assets/top-momoiro.jpg',
            cuisine: 'Asian',
            id: 1,
            location: 'Kota Kasablanka, Tebet, Jakarta',
            name: 'Momoiro',
            openingHours: '10.00 - 22.00',
            phoneNumber: '08980780780',
            rating: 4.2,
            slides: [
              'http://localhost:8080/assets/momoiro-1.jpg',
              'http://localhost:8080/assets/momoiro-2.jpg',
              'http://localhost:8080/assets/momoiro-3.jpg',
            ],
            slug: 'momoiro-kota-kasablanka',
          },
        ],
      },
    };
    const wrapper = shallow(<Hero {...props} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
