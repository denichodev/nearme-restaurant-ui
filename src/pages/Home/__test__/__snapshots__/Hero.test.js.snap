// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders correctly 1`] = `
<Row>
  <Col
    sm={12}
    xs={12}
  >
    <h1
      className="popularHeading"
    >
      Popular
    </h1>
  </Col>
  <Col
    md={5}
    sm={12}
    xs={12}
  >
    <div
      className="top"
    >
      <img
        alt="Momoiro"
        src="http://localhost:8080/assets/top-momoiro.jpg"
      />
      <Link
        replace={false}
        to="/r/momoiro-kota-kasablanka"
      >
        <div
          className="wrapper"
        >
          <Rating
            big={true}
            point={4.2}
          />
          <div
            className="cuisine"
          >
            Asian
          </div>
          <div
            className="name"
          >
            Momoiro
          </div>
          <div
            className="address"
          >
            Kota Kasablanka, Tebet, Jakarta
          </div>
        </div>
      </Link>
    </div>
  </Col>
  <Col
    md={7}
    sm={12}
    xs={12}
  >
    <Row />
  </Col>
</Row>
`;
